rpcs3/Emu/SysCalls/Modules/sys_mempool.cpp

```diff
@@ -1,40 +1,129 @@
 #include "stdafx.h"
 #include "Emu/Memory/Memory.h"
 #include "Emu/System.h"
+#include "Emu/IdManager.h"
 #include "Emu/SysCalls/Modules.h"
 
 #include "sysPrxForUser.h"
 
 extern Module sysPrxForUser;
 
+using sys_mempool_t = u32;
+
+struct memory_pool_t
```

Comment:
Hmm but the typedef is private, the pimpl idiom exists to hide all private parts of a class and reduce build times, overall across every class in a very large project it actually does add up as everyone who includes the header takes the parsing hit for something they won't ever use. 

I don't know - exposing implementation details on purpose feels wrong :).